OptionalAndList/OptionalAndList/ViewController.swift

```diff
@@ -0,0 +1,291 @@
+//
+//        linkedList.delLast()
+//        print(linkedList)
+        
+        print("Queue")
+        
+        let queue = Queue(value: 1)
+        queue.queue(value: 3)
+        queue.queue(value: 5)
+        queue.queue(value: 7)
+        queue.queue(value: 10)
+        print(queue.toString())
+        
+        if let a = queue.dequeue() {
+            print (a)
+        }
+        print(queue.toString())
+        
+        if let a = queue.dequeue() {
+            print (a)
+        }
+        print(queue.toString())
+        
+        if let a = queue.dequeue() {
+            print (a)
+        }
+        print(queue.toString())
+        
+        
+        print("Stack")
+        
+        let stack = Stack(value: 1)
+        stack.push(value: 3)
+        stack.push(value: 5)
+        stack.push(value: 7)
+        stack.push(value: 10)
+        print(stack.toString())
+        
+        if let a = stack.pop() {
+            print (a)
+        }
+        print(stack.toString())
+        
+        if let a = stack.pop() {
+            print (a)
+        }
+        print(stack.toString())
+        
+        if let a = stack.pop() {
+            print (a)
+        }
+        print(stack.toString())
+    }
+
+
+}
+
+enum OptionalError: Error{
+    case NilExeption
+}
+
+enum MyOptional<T> {
+    case some(T)
+    case none
+    
+    init(_ value: T){
+        self = .some(value)
+    }
+    
+    init(){
+        self = .none
+    }
+    
+    func unwrap() throws -> T{
+        switch self {
+        case .some(let x):
+            return x
+        default:
+            break
+        }
+        throw OptionalError.NilExeption
+    }
+}
+
+func add(_ first: MyOptional<Int>, _ second: MyOptional<Int>) throws -> MyOptional<Int>{
+    let a = try first.unwrap()
+    let b = try second.unwrap()
+    return MyOptional(a + b)
+}
+
+class Node<T>{
+    var value: T
+    var prev: Node?
+    var next: Node?
+    
+    init(value: T, prev: Node? = nil, next: Node? = nil){
+        self.value = value
+        self.prev = prev
+        self.next = next
+    }
+}
+
+
+class LinkedList<T>: CustomStringConvertible{
+    
+    var first: Node<T>?
+    var last: Node<T>?
+    
+//    init(){}
+    
+    init(value: T){
+        let firstNode = Node(value: value);
+        first = firstNode
+        last = firstNode
+    }
+    
+    func addFirst(value: T){
+        let newFirstNode = Node(value: value, next: first)
+        first = newFirstNode
+        if let oldFirst = newFirstNode.next {
+            oldFirst.prev = first
+        }
+    }
+    
+    func addLast(value: T){
+        let newLastNode = Node(value: value, prev: last)
+        last = newLastNode
+        if let oldLast = newLastNode.prev{
+            oldLast.next = last
+        }
+    }
+    
+    func delFirst(){
+        if first === last {
+            first = nil
+            last = nil
+        } else if let delFirstNode = first{
+        let firstNode = delFirstNode.next
```

Comment:
1. Don't use shorten words in names + Try to name more obvious names (`delFirstNode` -> `firstNodeToRemove`, `firstNode` -> `firstNodeCandidate`)
2. Try to refactor to avoid force unwrapping